### PR TITLE
raftstore-v2: split init may be out of dated when conf change ocurred

### LINKED ISSUE #14389: [Dynamic Regions] multiple rocksdb tikv panic for failed to accept snapshot
### What version of TiKV are you using?

[2023/03/11 13:53:44.306 +00:00] [INFO] [[lib.rs:86](http://lib.rs:86/)] ["Welcome to TiKV"]
[2023/03/11 13:53:44.306 +00:00] [INFO] [[lib.rs:91](http://lib.rs:91/)] ["Release Version:   6.7.0-alpha"]
[2023/03/11 13:53:44.306 +00:00] [INFO] [[lib.rs:91](http://lib.rs:91/)] ["Edition:           Community"]
[2023/03/11 13:53:44.306 +00:00] [INFO] [[lib.rs:91](http://lib.rs:91/)] ["Git Commit Hash:   fc9cf096b3eddc48e2f1df09dd6d716198067410"]
[2023/03/11 13:53:44.306 +00:00] [INFO] [[lib.rs:91](http://lib.rs:91/)] ["Git Commit Branch: heads/refs/tags/v6.7.0-alpha"]
[2023/03/11 13:53:44.306 +00:00] [INFO] [[lib.rs:91](http://lib.rs:91/)] ["UTC Build Time:    Unknown (env var does not exist when building)"]
[2023/03/11 13:53:44.306 +00:00] [INFO] [[lib.rs:91](http://lib.rs:91/)] ["Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)"]
[2023/03/11 13:53:44.306 +00:00] [INFO] [[lib.rs:91](http://lib.rs:91/)] ["Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure"]
[2023/03/11 13:53:44.306 +00:00] [INFO] [[lib.rs:91](http://lib.rs:91/)] ["Profile:           dist_release"]
### Steps to reproduce

```
[2023/03/11 13:55:35.837 +00:00] [FATAL] [[lib.rs:497](http://lib.rs:497/)] ["failed to accept snapshot [res=Ok(())] [accept_snapshot=false] [peer_id=2908] [region_id=2828]"] [backtrace="   0: tikv_util::set_panic_hook::{{closure}}\n             at /home/jenkins/agent/workspace/build-common/go/src/[github.com/pingcap/tikv/components/tikv_util/src/lib.rs:496:18\n](http://github.com/pingcap/tikv/components/tikv_util/src/lib.rs:496:18/n)   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/[boxed.rs:2032](http://boxed.rs:2032/):9\n      std::panicking::rust_panic_with_hook\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/[panicking.rs:692](http://panicking.rs:692/):13\n   2: std::panicking::begin_panic_handler::{{closure}}\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/[panicking.rs:579](http://panicking.rs:579/):13\n   3: std::sys_common::backtrace::__rust_end_short_backtrace\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/[backtrace.rs:137](http://backtrace.rs:137/):18\n   4: rust_begin_unwind\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/[panicking.rs:575](http://panicking.rs:575/):5\n   5: core::panicking::panic_fmt\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/[panicking.rs:65](http://panicking.rs:65/):14\n   6: raftstore_v2::operation::command::admin::split::<impl raftstore_v2::raft::peer::Peer<EK,ER>>::on_split_init\n             at /home/jenkins/agent/workspace/build-common/go/src/[github.com/pingcap/tikv/components/raftstore-v2/src/operation/command/admin/split.rs:720:13\n](http://github.com/pingcap/tikv/components/raftstore-v2/src/operation/command/admin/split.rs:720:13/n)      raftstore_v2::fsm::peer::PeerFsmDelegate<EK,ER,T>::on_msgs\n             at /home/jenkins/agent/workspace/build-common/go/src/[github.com/pingcap/tikv/components/raftstore-v2/src/fsm/peer.rs:266:44\n](http://github.com/pingcap/tikv/components/raftstore-v2/src/fsm/peer.rs:266:44/n)   7: <raftstore_v2::batch::store::StorePoller<EK,ER,T> as batch_system::batch::PollHandler<raftstore_v2::fsm::peer::PeerFsm<EK,ER>,raftstore_v2::fsm::store::StoreFsm>>::handle_normal\n             at /home/jenkins/agent/workspace/build-common/go/src/[github.com/pingcap/tikv/components/raftstore-v2/src/batch/store.rs:230:9\n](http://github.com/pingcap/tikv/components/raftstore-v2/src/batch/store.rs:230:9/n)      batch_system::batch::Poller<N,C,Handler>::poll\n             at /home/jenkins/agent/workspace/build-common/go/src/[github.com/pingcap/tikv/components/batch-system/src/batch.rs:422:27\n](http://github.com/pingcap/tikv/components/batch-system/src/batch.rs:422:27/n)   8: batch_system::batch::BatchSystem<N,C>::start_poller::{{closure}}\n             at /home/jenkins/agent/workspace/build-common/go/src/[github.com/pingcap/tikv/components/batch-system/src/batch.rs:550:17\n](http://github.com/pingcap/tikv/components/batch-system/src/batch.rs:550:17/n)      <std::thread::Builder as tikv_util::sys::thread::StdThreadBuildWrapper>::spawn_wrapper::{{closure}}\n             at /home/jenkins/agent/workspace/build-common/go/src/[github.com/pingcap/tikv/components/tikv_util/src/sys/thread.rs:427:23\n](http://github.com/pingcap/tikv/components/tikv_util/src/sys/thread.rs:427:23/n)      std::sys_common::backtrace::__rust_begin_short_backtrace\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/[backtrace.rs:121](http://backtrace.rs:121/):18\n   9: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/[mod.rs:551](http://mod.rs:551/):17\n      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panic/[unwind_safe.rs:271](http://unwind_safe.rs:271/):9\n      std::panicking::try::do_call\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/[panicking.rs:483](http://panicking.rs:483/):40\n      std::panicking::try\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/[panicking.rs:447](http://panicking.rs:447/):19\n      std::panic::catch_unwind\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/[panic.rs:137](http://panic.rs:137/):14\n      std::thread::Builder::spawn_unchecked_::{{closure}}\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/[mod.rs:550](http://mod.rs:550/):30\n      core::ops::function::FnOnce::call_once{{vtable.shim}}\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/[function.rs:513](http://function.rs:513/):5\n  10: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/[boxed.rs:2000](http://boxed.rs:2000/):9\n      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/[boxed.rs:2000](http://boxed.rs:2000/):9\n      std::sys::unix::thread::Thread::new::thread_start\n             at /rust/toolchains/nightly-2022-11-15-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys/unix/[thread.rs:108](http://thread.rs:108/):17\n  11: start_thread\n  12: clone\n"] [location=/home/jenkins/agent/workspace/build-common/go/src/[github.com/pingcap/tikv/components/raftstore-v2/src/operation/command/admin/split.rs:720](http://github.com/pingcap/tikv/components/raftstore-v2/src/operation/command/admin/split.rs:720)] [thread_name=rs-3-0]
```

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Close #14389

What's Changed:

```commit-message
split init may be out of dated
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore-v2/src/operation/command/admin/split.rs [hunk 1]
```diff
@@ -692,7 +697,20 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
...
+
+        // If peer_id in `split_init` is less than the current peer_id, the conf change
+        // for the peer should have occurred and we should just report finish to
+        // the source region of this out of dated peer initialization.
+        if self.storage().is_initialized() && self.persisted_index() >= RAFT_INIT_LOG_INDEX
+            || peer_id < self.peer().get_id()
+        {
             // Race with split operation. The tablet created by split will eventually be
             // deleted. We don't trim it.
             report_split_init_finish(store_ctx, split_init.derived_region_id, region_id, true);
```
Comment (BusyJay) [inline components/raftstore-v2/src/operation/command/admin/split.rs:716]: Better remove the tablet immediately as we don't implement gc anymore.

Comment (SpadeA-Tang) [reply components/raftstore-v2/src/operation/command/admin/split.rs:716]: report_split_init_finish will remove the tablet.

**Review by BusyJay — APPROVED** [review]

**Pull request merged**